runtime/rococo/src/propose_parachain.rs

```diff
@@ -200,11 +204,17 @@ decl_module! {
 			);
 			ensure!(validation_code.0.starts_with(runtime_common::WASM_MAGIC), Error::<T>::DefinitelyNotWasm);
 
-			let active_validators = Session::<T>::validators();
+			let mut active_validators = Session::<T>::validators();
+			ValidatorsToRetire::<T>::take().iter().for_each(|v| {
```

Comment:
You take them here and don't put the validators to retire back.

Reply:
Why is that correct? The retirement takes place on new session. Could you elaborate why we take it here?